### PR TITLE
Add checksum to the raw datasets

### PR DESCRIPTION
**Pull request by zhangguanheng66**
Update raw datasets with:

- checksum md5
- NUM_LINES

### CODE REVIEW

File: test/data/test_builtin_datasets.py [hunk 1]
```diff
@@ -65,7 +65,7 @@ def test_wikitext2(self):
         # Add test for the subset of the standard datasets
         train_dataset, test_dataset = torchtext.experimental.datasets.raw.WikiText2(data_select=('train', 'test'))
         self._helper_test_func(len(train_dataset), 36718, next(iter(train_dataset)), ' \n')
-        self._helper_test_func(len(test_dataset), 36718, next(iter(test_dataset)), ' \n')
+        self._helper_test_func(len(test_dataset), 4358, next(iter(test_dataset)), ' \n')
```
Comment (cpuhrsch) [inline test/data/test_builtin_datasets.py:68]: We just need to spread this to the other datasets as well and then this is good to go

**Review by cpuhrsch — APPROVED** [review]

**Review by cpuhrsch — APPROVED** [review]

**Comment by mthrok** [comment]
Question: Was there a discussion on the justification of the use of MD5? MD5 is fast but known to easily collide especially when two strings are similar. I know this might not be an immediate threat but wonder what you think.

https://www.mscs.dal.ca/~selinger/md5collision/

**Comment by mthrok** [comment]
Also, I am wondering when checksum is computed. (not immediately clear from the CC)
Does it happen at download time? at the time dataset object is instantiated? or every time files are loaded?

**Comment by zhangguanheng66** [comment]
> Also, I am wondering when checksum is computed. (not immediately clear from the CC)
> Does it happen at download time? at the time dataset object is instantiated? or every time files are loaded?

It's calculated after downloading.

**Comment by zhangguanheng66** [comment]
> Also, I am wondering when checksum is computed. (not immediately clear from the CC)
> Does it happen at download time? at the time dataset object is instantiated? or every time files are loaded?

We have a follow-up PR to switch to SHA2.

**Comment by zhangguanheng66** [comment]
> Also, I am wondering when checksum is computed. (not immediately clear from the CC)
> Does it happen at download time? at the time dataset object is instantiated? or every time files are loaded?

We have a follow-up PR to switch to SHA2 after benchmarking the performance of those two checksum.

**Comment by mthrok** [comment]
> > Also, I am wondering when checksum is computed. (not immediately clear from the CC)
> > Does it happen at download time? at the time dataset object is instantiated? or every time files are loaded?
> 
> We have a follow-up PR to switch to SHA2 after benchmarking the performance of those two checksum.

sounds good. thanks for the response.

**Pull request merged**